### PR TITLE
Bring auth.js coverage to 100%

### PR DESCRIPTION
**Pull request by anthonyringoet**
Issue: https://github.com/fastify/fastify-auth/issues/98

image

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by zekth — APPROVED** [review]
:v: nice!

File: test/auth.test.js [hunk 1]
```diff
@@ -27,6 +27,94 @@ test('Clean status code through auth pipeline', t => {
...
+test('Options: faulty run', t => {
+  t.plan(3)
+
+  const app = Fastify()
+  app.register(fastifyAuth).after(() => {
+    try {
+      app.addHook('preHandler', app.auth([successWithCode('one', 201)], { run: 'foo' }))
+      app.get('/', (req, res) => res.send(42))
+    } catch (error) {
+      t.ok(error)
```
Comment (Eomm) [inline test/auth.test.js:105]: What is the error?

File: test/auth.test.js [hunk 2]
```diff
@@ -27,6 +27,94 @@ test('Clean status code through auth pipeline', t => {
...
+    t.error(err)
+    t.equal(res.statusCode, 404)
+  })
+})
+
+test('Options: faulty run', t => {
+  t.plan(3)
+
+  const app = Fastify()
+  app.register(fastifyAuth).after(() => {
```
Comment (Eomm) [inline test/auth.test.js:100]: Is the error set here?

```suggestion
  app.register(fastifyAuth).after((err) => {
```

**Review by mcollina — APPROVED** [review]
lgtm

File: test/auth.test.js (hunk 1 above)
Comment (anthonyringoet) [reply test/auth.test.js:108]: Added extra check for `error.message`

File: test/auth.test.js (hunk 2 above)
Comment (anthonyringoet) [reply test/auth.test.js:103]: It doesn't seem like it. At that point, the fastify app is only decorated with `auth` (see `checkAuth` function in `auth.js`) and the underlying error gets thrown once the request is injected.

I might be missing something though 😄

Comment (Eomm) [reply test/auth.test.js:103]: I was trying to understand if the plugin is triggering the error or the `app.auth()` function

**Review by Eomm — APPROVED** [review]

**Pull request merged**